src/pages/membership/components/createYouthProfile/CreateYouthProfile.tsx

```diff
@@ -107,47 +86,76 @@ function CreateYouthProflle({ tunnistamoUser }: Props) {
       },
     };
 
-    createProfile({ variables })
-      .then(result => {
-        if (result.data) {
-          addServiceConnection({ variables: connectionVariables }).catch(
-            (error: Error) => {
-              Sentry.captureException(error);
-              setShowNotification(true);
-            }
-          );
-        }
-      })
-      .catch((error: Error) => {
+    addServiceConnection({ variables: connectionVariables }).catch(
+      (error: Error) => {
         Sentry.captureException(error);
         setShowNotification(true);
-      });
+      }
+    );
+  };
+
+  const handleOnValues = (formValues: FormValues) => {
+    const variables: CreateMyProfileVariables = getMutationVariables(
+      formValues,
+      data
+    );
+
+    if (data?.myProfile) {
+      updateProfile({ variables })
+        .then(result => {
+          if (!!result.data) connectService();
+        })
+        .catch((error: Error) => {
+          Sentry.captureException(error);
+          setShowNotification(true);
+        });
+    } else {
+      createProfile({ variables })
+        .then(result => {
+          if (!!result.data) connectService();
+        })
+        .catch((error: Error) => {
+          Sentry.captureException(error);
+          setShowNotification(true);
+        });
+    }
   };
+  // These allow us to set initial value of languageAtHome & profileLanguage
+  // to users current language.
+  const currentLangForProfile: Language = t('registration.language');
+  const currentLangForYouth: YouthLanguage = t('registration.language');
```

Comment:
The `useTranslation` hook returns the `i18n` instance that contains information about current language/locale. You can use this information to get the current language in the format you would like.

Pseudocode:
```js
function formatLocal(locale) {
  switch(locale) {
    case 'fi':
      return 'FINNISH';
    default:
      return 'FINNISH';
}
```

Reply:
I removed languages from translation files and created a helper for getting right value.